modules/33acrossBidAdapter.js

```diff
@@ -208,31 +198,103 @@ function _createServerRequest({bidRequest, gdprConsent = {}, uspConsent, pageUrl
   return {
     'method': 'POST',
     'url': url,
-    'data': JSON.stringify(contributeViewability(ttxRequest)),
+    'data': JSON.stringify(ttxRequest),
     'options': options
   }
 }
 
-// **************************** BUILD REQUESTS: BIDFLOORS *************************** //
-function _getBidFloors(getFloor, size) {
+// BUILD REQUESTS: SIZE INFERENCE
+function _transformSizes(sizes) {
+  if (utils.isArray(sizes) && sizes.length === 2 && !utils.isArray(sizes[0])) {
+    return [ _getSize(sizes) ];
+  }
+
+  return sizes.map(_getSize);
+}
+
+function _getSize(size) {
+  return {
+    w: parseInt(size[0], 10),
+    h: parseInt(size[1], 10)
+  }
+}
+
+// BUILD REQUESTS: PRODUCT INFERENCE
+function _getProduct(bidRequest) {
+  const { params, mediaTypes } = bidRequest;
+  const videoAdUnit = utils.deepAccess(bidRequest, 'mediaTypes.video', {});
+  const { context } = videoAdUnit;
+
+  const { banner, video } = mediaTypes;
+
+  if ((video && !banner) && context === 'instream') {
+    return PRODUCT.INSTREAM;
+  }
+
+  return (params.productId === PRODUCT.INVIEW) ? (params.productId) : PRODUCT.SIAB;
```

Comment:
whats the idea behind this `_getProduct()`? is it for validation? if not, then batter to just do `prod: params.productId`. if yes, then should here have a stricter check (eg. `PRODUCT.hasownproperty(params.productId)`) and then reject the request when it fail

Reply:
The idea is to set overrides and defaults. So if context is `instream` then the product has to be `instream` regardless of what the publisher sets in bidder param. The default is `siab` unless the product is `inview` in an outstream context. Which is what `_getProduct()` does.